### PR TITLE
[flutter_tools] remove unused bitcode validation code

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This code is only used in tests (should it be?) delete and remove tests.

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/bitcode.dart
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/bitcode.dart:6]: Can we move this into aot.dart and delete this file?

**Review by jmagman — COMMENTED** [review]
The usage was introduced in #36471 and removed in https://github.com/flutter/flutter/pull/52896/files#diff-4703aba3c9b446cc0d2fec973e3d3afaL40.
We check that the Flutter.framework contains bitcode in https://github.com/flutter/flutter/blob/a3b9e20a6b2f8ac2b2b4c2f032ee7478ed06e1de/dev/devicelab/bin/tasks/ios_content_validation_test.dart#L86 so I think we're covered.

File: packages/flutter_tools/lib/src/ios/bitcode.dart
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/ios/bitcode.dart]: Done

**Review by dnfield — CHANGES_REQUESTED** [review]
This code validated whether an appropriate/compatible version of Xcode was used compared to what we built the framework with. Where does that get checked now?

**Comment by jonahwilliams** [comment]
> This code validated whether an appropriate/compatible version of Xcode was used compared to what we built the framework with. Where does that get checked now?

It didn't, which is why I can remove it without changing the tool code.

**Comment by jonahwilliams** [comment]
Didn't we fix that by bumping the required xcode version anyway?

**Comment by dnfield** [comment]
Sorry, it _used_ to check that before some other change was made ahead of this.

We did bump the required Xcode version in our documentation, but this was the code that actually tested it. It's not clear to me if we're actually checking that anywhere now. @jmagman may know.

If not, we should be - this can cause issues for us if we, e.g., bump Xcode on the engine builders and not in the devicelab. We'd want the roll to fail in that case.

**Comment by jonahwilliams** [comment]
We do have a devicelab test that validates bitcode that @jmagman linked above. 

Rather than failing builds if a user Xcode and engine Xcode get out of sync, can we follow the example of the rest of the tool and automatically bump our doctor constraint?

**Comment by jonahwilliams** [comment]
Closing for now, I'll move the check into the appropriate place until we have a better solution

**Comment by dnfield** [comment]
The linked test doesn't validate the version of Xcode - I'm fine with doing that as a doctor check, but we should do it somewhere.

**Comment by jmagman** [comment]
Ah I see now it was actually testing the clang version, not that the framework was built with bitcode (it's coming back to me now, I remember issues with that version parsing when clang changed their format).  
As @jonahwilliams says, this mostly just covered by the existing Xcode minimum version check (which gets run on `flutter run` etc).  I guess technically you can swap out your toolchain and therefore clang version from under Xcode.

**Comment by dnfield** [comment]
Ahhh ok. I think it's fine to not cover the case of someone using a non-Xcode toolchain with Xcode - hopefully if they're doing that they know enough about what they're doing to make it all work out anyway. In that case this patch LGTM, unless there's now some other reason to not land it.

**Pull request closed without merging**
